Apache-Arrow-Flight-Tester/manager.py

```diff
@@ -1,50 +1,62 @@
+        decoded_result = result.body.to_pybytes().decode("utf-8")
 
-    action_body = encoded_node_url + encoded_node_mode
-    response = common.do_action(flight_client, "RegisterNode", action_body)
+        return decoded_result.split(";")
 
-    return response[0].body.to_pybytes()
+    def register_node(self, node_url: str, node_mode: Literal["cloud", "edge"]) -> list[Result]:
+        """Register a node with the given URL and mode in the manager."""
+        encoded_node_url = encode_argument(node_url)
+        encoded_node_mode = encode_argument(node_mode)
 
+        action_body = encoded_node_url + encoded_node_mode
+        response = self.do_action("RegisterNode", action_body)
 
-def remove_node(flight_client: FlightClient, node_url: str) -> list[Result]:
-    encoded_node_url = common.encode_argument(node_url)
+        return response[0].body.to_pybytes()
 
-    return common.do_action(flight_client, "RemoveNode", encoded_node_url)
+    def remove_node(self, node_url: str) -> list[Result]:
+        """Remove the node with the given URL from the manager"""
+        encoded_node_url = encode_argument(node_url)
 
+        return self.do_action("RemoveNode", encoded_node_url)
 
-def execute_query(flight_client: FlightClient, query: str) -> None:
-    print("Retrieving cloud node that can execute the query...")
-    query_descriptor = flight.FlightDescriptor.for_command(query)
-    flight_info = flight_client.get_flight_info(query_descriptor)
+    def query(self, query: str) -> None:
+        """Retrieve a cloud node that can execute the given query and execute the query on the node."""
+        print("Retrieving cloud node that can execute the query...")
+        query_descriptor = flight.FlightDescriptor.for_command(query)
+        flight_info = self.flight_client.get_flight_info(query_descriptor)
 
-    endpoint = flight_info.endpoints[0]
-    cloud_node_url = endpoint.locations[0]
+        endpoint = flight_info.endpoints[0]
+        cloud_node_url = endpoint.locations[0]
 
-    print(f"Executing query on {cloud_node_url}...")
-    cloud_client = flight.FlightClient(cloud_node_url)
-    common.do_get(cloud_client, endpoint.ticket)
+        print(f"Executing query on {cloud_node_url}...")
+        cloud_client = ModelarDBServerFlightClient(cloud_node_url)
+        cloud_client.do_get(endpoint.ticket)
 
 
 if __name__ == "__main__":
-    manager_client = flight.FlightClient("grpc://127.0.0.1:9998")
-    common.create_test_tables(manager_client)
+    manager_client = ModelarDBManagerFlightClient("grpc://127.0.0.1:9998")
+    manager_client.create_test_tables()
+
+    print(manager_client.initialize_database(["test_table_1"]))
+
+    print(manager_client.register_node("grpc://127.0.0.1:9999", "edge"))
+    print(manager_client.remove_node("grpc://127.0.0.1:9999"))
 
-    execute_query(manager_client, "SELECT * FROM test_model_table_1 LIMIT 5")
+    manager_client.query("SELECT * FROM test_model_table_1 LIMIT 5")
```

Comment:
It seems `query()` is executed without first registering a cloud node with the manager for executing the query, if this is on purpose, maybe add a comment that documents why this is done.

Reply:
Added text to docstring that explains that it is assumed that at least one cloud node is already registered.